roles/ansible/tower/manage-credential-types/tasks/process-credential-type.yml

```diff
@@ -0,0 +1,59 @@
+---
+
+- name: "Get the credential_type id based on the name"
+  set_fact:
+    credential_type_id: "{{ item.id }}"
+  when:
+  - item.name|trim == credential_type.name|trim
+  with_items:
+  - "{{ existing_credential_types_output.rest_output }}"
+
+- name: create temporary directory to store the processed template
+  tempfile:
+    state: directory
+    suffix: cred_type
+  register: tmpdir
+
+- name: "Process Credential Type Template"
```

Comment:
This is a lot of extra overhead processing the template by first writing it a file, then slurping the file then passing it to the uri module. I'd suggest just using the `lookup('template' ...` option for the `body` as part of the `uri` call. See example here: https://github.com/redhat-cop/infra-ansible/blob/master/roles/ansible/tower/manage-credentials/tasks/process-credential.yml#L23

BTW: these many steps may have been so there's a "file to look at", but this can easily be done with a verbose level set correctly to see the input/output.

Reply:
Also, the current code leaves temporary directories/files behind, which moving to using the `lookup('template', ...` will also solve

Reply:
Changed with your proposal, but I still prefer the other approach where you can actually debug the file generated from the template and examine it in case something fails. Removing the temporary file is just another play, so that can be achieved as well

Reply:
@makentenza ok, great. So since line 30 now uses `lookup('template', ...` please remove these 2 tasks  -i.e.: creating the temp file and setting the fact